omod/src/main/java/org/openmrs/web/rest/resource/TagResource.java

```diff
@@ -0,0 +1,139 @@
+        "1.12.*", "2.0.*", "2.1.*" })
+public class TagResource extends DelegatingCrudResource<Tag> {
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#newDelegate()
+	 */
+	@Override
+	public Tag newDelegate() {
+		return new Tag();
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getByUniqueId(String)
+	 */
+	@Override
+	public Tag getByUniqueId(String uuid) {
+		return getService().getTagByUuid(uuid);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#save(Object)
+	 */
+	@Override
+	public Tag save(Tag tag) {
+		return getService().saveTag(tag);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#delete(Object,String, RequestContext)
+	 */
+	@Override
+	protected void delete(Tag tag, String reason, RequestContext context) {
+		throw new ResourceDoesNotSupportOperationException("delete not allowed on tag");
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#purge(Object, RequestContext)
+	 */
+	@Override
+	public void purge(Tag tag, RequestContext context) {
+		getService().purgeTag(tag);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getRepresentationDescription(Representation)
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = null;
+		if (rep instanceof DefaultRepresentation || rep instanceof RefRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("tag");
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		} else if (rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("tag");
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+			description.addProperty("auditInfo");
+			description.addSelfLink();
+		}
+		return description;
+	}
+	
+	/**
+	 * @return the TagService
+	 */
+	private TagService getService() {
+		return Context.getService(TagService.class);
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		
+		description.addRequiredProperty("tag");
+		description.addRequiredProperty("objectType");
+		description.addRequiredProperty("objectUuid");
+		
+		return description;
+	}
+	
+	@Override
+	protected NeedsPaging<Tag> doGetAll(RequestContext context) throws ResponseException {
+		return new NeedsPaging<Tag>(Context.getService(TagService.class).getTags(""), context);
+	}
+	
+	@Override
+	protected PageableResult doSearch(RequestContext context) {
+		String tagName = context.getRequest().getParameter("tag");
+		if (tagName != null) {
```

Comment:
Use StringUtils.isNotBlank otherwise `tagName` could be an empty string or white space character